examples/sdl-audio.rs

```diff
@@ -0,0 +1,135 @@
+const SAMPLES_PER_PERIOD: f32 = SAMPLE_RATE as f32 * PERIOD;
+const PITCH_CHANGE_PER_SAMPLE: f32 = (PITCH_MAX - PITCH_MIN) / SAMPLES_PER_PERIOD;
+
+fn main() -> Result<(), core::convert::Infallible> {
+    // Prepare the audio "engine" with gate control
+    let gate = Arc::new(AtomicBool::new(false));
+    let audio_wrapper = AudioWrapper::new(gate.clone());
+
+    let audio_spec = AudioSpecDesired {
+        freq: Some(SAMPLE_RATE),
+        channels: Some(1),
+        samples: Some(32),
+    };
+
+    // This is the initialisation of SDL and capturing its audio device.
+    let audio_device = sdl2::init()
+        .unwrap()
+        .audio()
+        .unwrap()
+        .open_playback(None, &audio_spec, |_| audio_wrapper)
+        .unwrap();
+
+    audio_device.resume();
+
+    let output_settings = OutputSettingsBuilder::new()
+        .scale(4)
+        .theme(embedded_graphics_simulator::BinaryColorTheme::OledWhite)
+        .build();
+
+    // This is the window initialisation. It calls the same SDL init function inside, which
+    // is allowed as long as the init calls are from the same OS thread.
+    let mut window = Window::new("Simulator audio example", &output_settings);
+
+    let text_style = MonoTextStyle::new(&FONT_6X10, BinaryColor::On);
+    let text_position = Point::new(25, 30);
+    let text = Text::new("Press space...", text_position, text_style);
+
+    let mut display: SimulatorDisplay<BinaryColor> = SimulatorDisplay::new(Size::new(128, 64));
+    text.draw(&mut display).unwrap();
+    'running: loop {
+        window.update(&display);
+
+        for event in window.events() {
+            match event {
+                SimulatorEvent::Quit => break 'running,
+                SimulatorEvent::KeyDown {
+                    keycode, repeat, ..
+                } if keycode == Keycode::Space && !repeat => {
+                    gate.store(true, Ordering::SeqCst);
+                    display.clear(BinaryColor::On).unwrap();
+                }
+                SimulatorEvent::KeyUp { keycode, .. } => match keycode {
+                    Keycode::Space => {
+                        gate.store(false, Ordering::SeqCst);
+                        display.clear(BinaryColor::Off).unwrap();
+                        text.draw(&mut display).unwrap();
+                    }
+                    _ => {}
+                },
+                _ => {}
+            }
+        }
+    }
+
+    Ok(())
+}
+
+struct AudioWrapper {
+    gate: Arc<AtomicBool>,
+    phase: f32,
+    pitch: f32,
+}
+
+impl AudioWrapper {
+    fn new(gate: Arc<AtomicBool>) -> Self {
+        Self {
+            gate,
+            phase: 0.0,
+            pitch: PITCH_MIN,
+        }
+    }
+}
+
+impl AudioCallback for AudioWrapper {
+    type Channel = f32;
+
+    fn callback(&mut self, out: &mut [f32]) {
+        let gate = self.gate.load(Ordering::SeqCst);
+        if !gate {
+            out.fill(0.0);
+            return;
+        }
```

Comment:
Not necessary, but I think it's a bit nicer to always start at minimum pitch.
```suggestion
        if !gate {
            self.pitch = PITCH_MIN;
            out.fill(0.0);
            return;
        }
```

Reply:
allright :)